tests/single_node_test.go

```diff
@@ -0,0 +1,74 @@
+package tests
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"kubevirt.io/controller-lifecycle-operator-sdk/pkg/sdk/api"
+	sspv1beta1 "kubevirt.io/ssp-operator/api/v1beta1"
+)
+
+var _ = Describe("Single Node Topology", func() {
+	BeforeEach(func() {
+		strategy.SkipUnlessSingleReplicaTopologyMode()
+		waitUntilDeployed()
+	})
+
+	AfterEach(func() {
+		strategy.RevertToOriginalSspCr()
+	})
+
+	It("[test_id:7792] Number of Template Validator replicas is not bigger then one", func() {
+		watch, err := StartWatch(sspListerWatcher)
+		Expect(err).ToNot(HaveOccurred())
+		defer watch.Stop()
+
+		var newValidatorReplicas int32 = 3
+		updateSsp(func(foundSsp *sspv1beta1.SSP) {
+			foundSsp.Spec.TemplateValidator.Replicas = &newValidatorReplicas
+		})
+
+		// Watch changes until above change
+		err = WatchChangesUntil(watch, func(updatedSsp *sspv1beta1.SSP) bool {
+			return *updatedSsp.Spec.TemplateValidator.Replicas == newValidatorReplicas &&
+				updatedSsp.Generation > updatedSsp.Status.ObservedGeneration
+		}, shortTimeout)
+		Expect(err).ToNot(HaveOccurred())
+
+		// Watch changes until SSP operator updates ObservedGeneration
+		err = WatchChangesUntil(watch, func(updatedSsp *sspv1beta1.SSP) bool {
+			return *updatedSsp.Spec.TemplateValidator.Replicas == newValidatorReplicas &&
+				updatedSsp.Generation == updatedSsp.Status.ObservedGeneration && updatedSsp.Status.Phase == api.PhaseDeployed
+		}, shortTimeout)
+		Expect(err).ToNot(HaveOccurred())
+
+		deployment := getTemplateValidatorDeployment()
+		Expect(int(deployment.Status.Replicas)).Should(Equal(1), "In Single Mode Topology the number of replicas is at most 1")
+	})
+
+	It("[test_id:7793] Number of Template Validator replicas can be set to 0", func() {
+		watch, err := StartWatch(sspListerWatcher)
```

Comment:
This test case does not need to use watch. It is enough to call `waitUntilDeployed()` before checking the template validator deployment.

Watch on SSP is useful when the test needs to be sure that the operator reacted to a change, but there is no external way to find out. For example in the previous test, you set the number of replicas to 3, but because this is single node, the deployment is not actually changed. In that case the test needs to watch the `ObservedGeneration` to be sure that the operator has seen the new SSP CR.